### PR TITLE
DBZ-3632 Using correct configuration subset and SQL command when vali…

### PR DESCRIPTION
**Pull request by jcechace**
…dating connection

https://issues.redhat.com/browse/DBZ-3632

### CODE REVIEW

File: src/main/java/io/debezium/connector/db2/Db2Connector.java [hunk 1]
```diff
@@ -74,10 +74,10 @@ protected void validateConnection(Map<String, ConfigValue> configValues, Configu
         ConfigValue hostnameValue = configValues.get(RelationalDatabaseConnectorConfig.HOSTNAME.name());
         // Try to connect to the database ...
         Db2ConnectorConfig connectorConfig = new Db2ConnectorConfig(config);
-        try (Db2Connection connection = new Db2Connection(connectorConfig.getConfig())) {
+        try (Db2Connection connection = new Db2Connection(connectorConfig.jdbcConfig())) {
```
Comment (gunnarmorling) [inline src/main/java/io/debezium/connector/db2/Db2Connector.java:77]: You should be able to just do `config.jdbcConfig()`, i.e. no need for the new method.

**Review by gunnarmorling — COMMENTED** [review]
One comment inline.

File: src/main/java/io/debezium/connector/db2/Db2Connector.java (hunk 1 above)
Comment (jcechace) [reply src/main/java/io/debezium/connector/db2/Db2Connector.java:77]: Unfortunately, as I stated during our team call, the method is not part of ``RelationalDatabaseConnectorConfig```. Although I agree that it should probably be there. However, that would mean changes to pretty much all RDB connectors we have.

Personally, I have no problem moving it to the parent class. It's your call.

**Comment by gunnarmorling** [comment]
@jcechace getting back to this, sorry for the delay. Please check out PR https://github.com/debezium/debezium/pull/2521, which adds `RelationalDatabaseConnectorConfig.getJdbcConfig()`. Could you please rework and force-push this PR here once that other one has been merged. Thx!

**Comment by gunnarmorling** [comment]
@jcechace, main repo PR is merged, you can rework this one here now. Thx!

**Comment by jcechace** [comment]
@gunnarmorling  done

File: src/main/java/io/debezium/connector/db2/Db2Connector.java [hunk 2]
```diff
@@ -74,10 +74,10 @@ protected void validateConnection(Map<String, ConfigValue> configValues, Configu
         ConfigValue hostnameValue = configValues.get(RelationalDatabaseConnectorConfig.HOSTNAME.name());
         // Try to connect to the database ...
         Db2ConnectorConfig connectorConfig = new Db2ConnectorConfig(config);
-        try (Db2Connection connection = new Db2Connection(connectorConfig.getConfig())) {
+        try (Db2Connection connection = new Db2Connection(connectorConfig.getJdbcConfig())) {
             try {
                 connection.connect();
-                connection.execute("SELECT version()");
+                connection.execute("SELECT 1 FROM sysibm.sysdummy1;");
```
Comment (gunnarmorling) [inline src/main/java/io/debezium/connector/db2/Db2Connector.java:80]: @jcechace, does this table always exist? Can't you just do `SELECT 1`?

Comment (jcechace) [reply src/main/java/io/debezium/connector/db2/Db2Connector.java:80]: I'm not sure whether DB2 permits SELECT without a table. Anyway according to docs sysdummy1 should always exist.

https://www.ibm.com/docs/en/db2-for-zos/12?topic=tables-sysdummyx

**Comment by gunnarmorling** [comment]
Applied to master (=1.7) and 1.6. Thx, @jcechace!

**Pull request closed without merging**